File: slot.java
// slot.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.util.concurrent.ThreadLocalRandom;

public class slot {
    private static final String RESET = "\u001B[0m";
    private static final String RED = "\u001B[91m";
    private static final String GREEN = "\u001B[92m";
    private static final String YELLOW = "\u001B[93m";
    private static final String BLUE = "\u001B[94m";
    private static final String MAGENTA = "\u001B[95m";
    private static final String CYAN = "\u001B[96m";
    private static final String BOLD = "\u001B[1m";

    private static String colorize(String text, String color) {
        return color + text + RESET;
    }

    private static final String[] SYMBOLS = {"🍒", "🍋", "🍊", "🍇", "🔔", "💎", "7"};
    private static final String[] SYMBOL_COLORS = {RED, YELLOW, MAGENTA, MAGENTA, CYAN, CYAN, BOLD};
    private static final Map<String, Integer> SYMBOL_VALUES = new HashMap<>();
    static {
        SYMBOL_VALUES.put("🍒", 5);
        SYMBOL_VALUES.put("🍋", 5);
        SYMBOL_VALUES.put("🍊", 5);
        SYMBOL_VALUES.put("🍇", 8);
        SYMBOL_VALUES.put("🔔", 10);
        SYMBOL_VALUES.put("💎", 100);
        SYMBOL_VALUES.put("7", 50);
    }

    private static class Stats {
        int rolls;
        int total_won;
        int best_win;
        int balance;
    }

    private int balance;
    private int bet;
    private List<String[]> history = new ArrayList<>();
    private Stats stats;
    private String statsFile;
    private Scanner scanner;

    public slot(int initialBalance) {
        balance = initialBalance;
        statsFile = System.getProperty("user.home") + "/.slot_stats.json";
        loadStats();
        scanner = new Scanner(System.in);
    }

    private void loadStats() {
        stats = new Stats();
        try {
            String json = new String(Files.readAllBytes(Paths.get(statsFile)));
            // упрощённый парсинг
            stats.rolls = extractInt(json, "rolls");
            stats.total_won = extractInt(json, "total_won");
            stats.best_win = extractInt(json, "best_win");
            stats.balance = extractInt(json, "balance");
            if (stats.balance == 0 && stats.rolls == 0) stats.balance = 100;
        } catch (Exception e) {
            stats.balance = 100;
        }
        balance = stats.balance > 0 ? stats.balance : 100;
    }

    private int extractInt(String json, String key) {
        int idx = json.indexOf("\"" + key + "\"");
        if (idx == -1) return 0;
        int start = json.indexOf(":", idx) + 1;
        int end = json.indexOf(",", start);
        if (end == -1) end = json.indexOf("}", start);
        try { return Integer.parseInt(json.substring(start, end).trim()); } catch (Exception e) { return 0; }
    }

    private void saveStats() {
        stats.balance = balance;
        String json = "{\"rolls\":" + stats.rolls + ",\"total_won\":" + stats.total_won +
                      ",\"best_win\":" + stats.best_win + ",\"balance\":" + stats.balance + "}";
        try { Files.write(Paths.get(statsFile), json.getBytes()); } catch (IOException e) {}
    }

    private String[] spin(int betAmount) {
        if (betAmount > balance) {
            System.out.println(colorize("Недостаточно средств.", RED));
            return null;
        }
        if (betAmount <= 0) {
            System.out.println(colorize("Ставка должна быть положительной.", RED));
            return null;
        }
        bet = betAmount;
        balance -= bet;
        System.out.println(colorize("Вращение...", BOLD));
        // Анимация
        Random rnd = ThreadLocalRandom.current();
        for (int i=0; i<3; i++) {
            String[] temp = new String[3];
            for (int j=0; j<3; j++) temp[j] = SYMBOLS[rnd.nextInt(SYMBOLS.length)];
            System.out.printf("\r%s %s %s", temp[0], temp[1], temp[2]);
            try { Thread.sleep(200); } catch (InterruptedException e) {}
        }
        // Результат
        String[] result = new String[3];
        for (int i=0; i<3; i++) result[i] = SYMBOLS[rnd.nextInt(SYMBOLS.length)];
        System.out.printf("\r%s %s %s\n", result[0], result[1], result[2]);
        int win = calculateWin(result);
        balance += win;
        stats.rolls++;
        stats.total_won += win;
        if (win > stats.best_win) stats.best_win = win;
        history.add(result);
        if (win > 0) {
            System.out.println(colorize("Вы выиграли " + win + " монет!", GREEN));
        } else {
            System.out.println(colorize("Ничего не выиграно.", RED));
        }
        saveStats();
        return result;
    }

    private int calculateWin(String[] result) {
        if (result[0].equals("💎") && result[1].equals("💎") && result[2].equals("💎")) {
            return bet * 100;
        }
        if (result[0].equals(result[1]) && result[1].equals(result[2])) {
            return bet * SYMBOL_VALUES.get(result[0]);
        }
        // Два одинаковых + джокер
        for (String sym : SYMBOLS) {
            if (sym.equals("💎")) continue;
            int count = 0;
            for (String s : result) if (s.equals(sym)) count++;
            if (count == 2 && contains(result, "💎")) {
                return bet * 2 * SYMBOL_VALUES.get(sym);
            }
        }
        if (contains(result, "💎")) {
            return bet * 2;
        }
        return 0;
    }

    private boolean contains(String[] arr, String val) {
        for (String s : arr) if (s.equals(val)) return true;
        return false;
    }

    private void displayBalance() {
        System.out.println(colorize("Баланс: " + balance + " монет", YELLOW));
    }

    private void displayStats() {
        System.out.println(colorize("📊 Статистика:", BOLD));
        System.out.println("  Вращений: " + stats.rolls);
        System.out.println("  Всего выиграно: " + stats.total_won);
        System.out.println("  Лучший выигрыш: " + stats.best_win);
        System.out.println("  Текущий баланс: " + balance);
    }

    public void run() {
        System.out.println(colorize("🎰 Добро пожаловать в Слот-машину!", BOLD));
        System.out.println("Команды: spin <ставка>, balance, stats, deposit <сумма>, help, quit");
        displayBalance();
        while (true) {
            System.out.print("> ");
            String line = scanner.nextLine().trim();
            if (line.isEmpty()) continue;
            String[] parts = line.split("\\s+");
            String cmd = parts[0].toLowerCase();
            if (cmd.equals("quit") || cmd.equals("q")) {
                System.out.println("Выход.");
                saveStats();
                break;
            } else if (cmd.equals("balance") || cmd.equals("b")) {
                displayBalance();
            } else if (cmd.equals("stats")) {
                displayStats();
            } else if (cmd.equals("spin")) {
                if (parts.length != 2) {
                    System.out.println("Используйте: spin <ставка>");
                    continue;
                }
                try {
                    int betAmount = Integer.parseInt(parts[1]);
                    spin(betAmount);
                } catch (NumberFormatException e) {
                    System.out.println("Ставка должна быть числом.");
                }
            } else if (cmd.equals("deposit")) {
                if (parts.length != 2) {
                    System.out.println("Используйте: deposit <сумма>");
                    continue;
                }
                try {
                    int amount = Integer.parseInt(parts[1]);
                    if (amount <= 0) {
                        System.out.println("Сумма должна быть положительной.");
                        continue;
                    }
                    balance += amount;
                    System.out.println(colorize("Баланс пополнен на " + amount + ". Текущий баланс: " + balance, GREEN));
                    saveStats();
                } catch (NumberFormatException e) {
                    System.out.println("Сумма должна быть числом.");
                }
            } else if (cmd.equals("help") || cmd.equals("h")) {
                System.out.println("Команды:");
                System.out.println("  spin <ставка>  - сделать ставку и крутить барабаны");
                System.out.println("  balance        - показать баланс");
                System.out.println("  stats          - показать статистику");
                System.out.println("  deposit <сумма> - пополнить баланс");
                System.out.println("  quit           - выход");
            } else {
                System.out.println("Неизвестная команда. Введите 'help' для справки.");
            }
        }
        scanner.close();
    }

    public static void main(String[] args) {
        slot game = new slot(100);
        game.run();
    }
}

File: slot.cs
// slot.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;
using System.Threading;

class SlotMachine
{
    static string Colorize(string text, string color)
    {
        string col = color switch
        {
            "red" => "\x1b[91m",
            "green" => "\x1b[92m",
            "yellow" => "\x1b[93m",
            "blue" => "\x1b[94m",
            "magenta" => "\x1b[95m",
            "cyan" => "\x1b[96m",
            "bold" => "\x1b[1m",
            _ => "\x1b[0m"
        };
        return col + text + "\x1b[0m";
    }

    static string[] SYMBOLS = {"🍒", "🍋", "🍊", "🍇", "🔔", "💎", "7"};
    static string[] SYMBOL_COLORS = {"red", "yellow", "magenta", "magenta", "cyan", "cyan", "bold"};
    static Dictionary<string, int> SYMBOL_VALUES = new Dictionary<string, int>
    {
        {"🍒", 5}, {"🍋", 5}, {"🍊", 5}, {"🍇", 8}, {"🔔", 10}, {"💎", 100}, {"7", 50}
    };

    class Stats
    {
        public int rolls { get; set; }
        public int total_won { get; set; }
        public int best_win { get; set; }
        public int balance { get; set; }
    }

    private int balance;
    private int bet;
    private List<string[]> history = new List<string[]>();
    private Stats stats;
    private string statsFile;

    public SlotMachine(int initialBalance = 100)
    {
        balance = initialBalance;
        statsFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".slot_stats.json");
        LoadStats();
    }

    void LoadStats()
    {
        if (File.Exists(statsFile))
        {
            try
            {
                string json = File.ReadAllText(statsFile);
                stats = JsonSerializer.Deserialize<Stats>(json);
                if (stats != null)
                {
                    balance = stats.balance > 0 ? stats.balance : 100;
                }
                else stats = new Stats { balance = 100 };
            }
            catch { stats = new Stats { balance = 100 }; }
        }
        else stats = new Stats { balance = 100 };
    }

    void SaveStats()
    {
        stats.balance = balance;
        string json = JsonSerializer.Serialize(stats);
        File.WriteAllText(statsFile, json);
    }

    string[] Spin(int betAmount)
    {
        if (betAmount > balance)
        {
            Console.WriteLine(Colorize("Недостаточно средств.", "red"));
            return null;
        }
        if (betAmount <= 0)
        {
            Console.WriteLine(Colorize("Ставка должна быть положительной.", "red"));
            return null;
        }
        bet = betAmount;
        balance -= bet;
        Console.WriteLine(Colorize("Вращение...", "bold"));
        // Анимация
        Random rnd = new Random();
        for (int i=0; i<3; i++)
        {
            string[] temp = new string[3];
            for (int j=0; j<3; j++) temp[j] = SYMBOLS[rnd.Next(SYMBOLS.Length)];
            Console.Write($"\r{temp[0]} {temp[1]} {temp[2]}");
            Thread.Sleep(200);
        }
        // Результат
        string[] result = new string[3];
        for (int i=0; i<3; i++) result[i] = SYMBOLS[rnd.Next(SYMBOLS.Length)];
        Console.Write($"\r{result[0]} {result[1]} {result[2]}\n");
        int win = CalculateWin(result);
        balance += win;
        stats.rolls++;
        stats.total_won += win;
        if (win > stats.best_win) stats.best_win = win;
        history.Add(result);
        if (win > 0)
            Console.WriteLine(Colorize($"Вы выиграли {win} монет!", "green"));
        else
            Console.WriteLine(Colorize("Ничего не выиграно.", "red"));
        SaveStats();
        return result;
    }

    int CalculateWin(string[] result)
    {
        if (result[0] == "💎" && result[1] == "💎" && result[2] == "💎")
            return bet * 100;
        if (result[0] == result[1] && result[1] == result[2])
            return bet * SYMBOL_VALUES[result[0]];
        // Два одинаковых + джокер
        foreach (string sym in SYMBOLS)
        {
            if (sym == "💎") continue;
            int count = 0;
            foreach (string s in result) if (s == sym) count++;
            if (count == 2 && Array.Exists(result, s => s == "💎"))
                return bet * 2 * SYMBOL_VALUES[sym];
        }
        if (Array.Exists(result, s => s == "💎"))
            return bet * 2;
        return 0;
    }

    void DisplayBalance()
    {
        Console.WriteLine(Colorize($"Баланс: {balance} монет", "yellow"));
    }

    void DisplayStats()
    {
        Console.WriteLine(Colorize("📊 Статистика:", "bold"));
        Console.WriteLine($"  Вращений: {stats.rolls}");
        Console.WriteLine($"  Всего выиграно: {stats.total_won}");
        Console.WriteLine($"  Лучший выигрыш: {stats.best_win}");
        Console.WriteLine($"  Текущий баланс: {balance}");
    }

    public void Run()
    {
        Console.WriteLine(Colorize("🎰 Добро пожаловать в Слот-машину!", "bold"));
        Console.WriteLine("Команды: spin <ставка>, balance, stats, deposit <сумма>, help, quit");
        DisplayBalance();
        while (true)
        {
            Console.Write("> ");
            string input = Console.ReadLine().Trim();
            if (input == "quit" || input == "q")
            {
                Console.WriteLine("Выход.");
                SaveStats();
                break;
            }
            else if (input == "balance" || input == "b")
            {
                DisplayBalance();
            }
            else if (input == "stats")
            {
                DisplayStats();
            }
            else if (input.StartsWith("spin "))
            {
                string[] parts = input.Split(' ');
                if (parts.Length != 2)
                {
                    Console.WriteLine("Используйте: spin <ставка>");
                    continue;
                }
                if (!int.TryParse(parts[1], out int betAmount))
                {
                    Console.WriteLine("Ставка должна быть числом.");
                    continue;
                }
                Spin(betAmount);
            }
            else if (input.StartsWith("deposit "))
            {
                string[] parts = input.Split(' ');
                if (parts.Length != 2)
                {
                    Console.WriteLine("Используйте: deposit <сумма>");
                    continue;
                }
                if (!int.TryParse(parts[1], out int amount) || amount <= 0)
                {
                    Console.WriteLine("Сумма должна быть положительным числом.");
                    continue;
                }
                balance += amount;
                Console.WriteLine(Colorize($"Баланс пополнен на {amount}. Текущий баланс: {balance}", "green"));
                SaveStats();
            }
            else if (input == "help" || input == "h")
            {
                Console.WriteLine("Команды:");
                Console.WriteLine("  spin <ставка>  - сделать ставку и крутить барабаны");
                Console.WriteLine("  balance        - показать баланс");
                Console.WriteLine("  stats          - показать статистику");
                Console.WriteLine("  deposit <сумма> - пополнить баланс");
                Console.WriteLine("  quit           - выход");
            }
            else
            {
                Console.WriteLine("Неизвестная команда. Введите 'help' для справки.");
            }
        }
    }

    static void Main()
    {
        SlotMachine game = new SlotMachine();
        game.Run();
    }
}
